clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go

```diff
@@ -146,18 +148,37 @@ func NewResourceUsageGatherer(c clientset.Interface, host string, port int, prov
 			}
 		}
 
+		for _, address := range options.MasterIPs {
+			nodesToConsider[address] = true
+		}
+		masterIPs := sets.NewString(options.MasterIPs...)
+		nodeNames := sets.NewString(masterIPs.List()...)
 		for _, node := range nodeList.Items {
-			if options.Nodes == AllNodes || masterNodes.Has(node.Name) || nodesToConsider[node.Name] {
+			nodeNames.Insert(node.Name)
```

Comment:
Looks like we're mixing node IPs and node names. I don't like it :)

Reply:
Good callout - also potentially may cause node to appear twice (if both appearing as node, and specified as ip). Will look into it.